🚀 Deploy Scuffbot to Kube

sam1357 opened the pull request:
Getting Scuffbot ready to be deployed to Kube.

This involves several changes, such as how the config and secrets are handled.

It also uses the new `discord-bot` helm chart from `infra-helm-charts`.

MasterOfCubesAU commented on infra/values.yaml:
```diff
@@ -0,0 +1,15 @@
+        BOT_TOKEN: /secrets/bot-token
+        CONFIG_FILE: /config/config.yaml
+      config:
+        fileVariable: "scuffbotConfig"
```
Here, could we not just have `config: placeholder` and then in the pipeline, substitute it with `--set-file config=config.yaml`?

sam1357 replied:
True, we could have it as config: <variable name>. The reason I'm giving the option to code in a variable is for the consideration if multiple containers need each their own config.yaml, if that makes sense

MasterOfCubesAU commented on launcher.py:
```diff
@@ -1,9 +1,15 @@
 
-bot = SCUFFBOT(args.dev)
-bot.run()
+with open(os.environ["APPLICATION_IDS"], "r", encoding="utf-8") as f:
```
Do you think there would be a point having a check here to show a friendly message when this env var does not result in a file? Or do you think the `open()` call raising an exception as a result of `local-secrets` not existing is sufficient?

sam1357 replied:
Couldn't hurt

MasterOfCubesAU replied:
Add to this PR? Or nah

sam1357 replied:
Probs in another PR - I think Scuffbot needs a lot of rework anyway lol

MasterOfCubesAU reviewed (commented):
See comment

MasterOfCubesAU commented on .env:
```diff
@@ -0,0 +1,3 @@
+APPLICATION_ID=.local-secrets/application-id
+BOT_TOKEN=.local-secrets/bot-tokena
```
typo

sam1357 committed:
🚀 Deploy Scuffbot to Kube

Getting Scuffbot ready to be deployed to Kube.

This involves several changes, such as how the config and secrets are handled.

It also uses the new `discord-bot` helm chart from `infra-helm-charts`.

MasterOfCubesAU commented on config.yaml:
Update to include prod config pls

MasterOfCubesAU commented on db/.env.template:
```diff
@@ -1,4 +1,4 @@
-MYSQL_PASSWORD=aStrongPassword
-MYSQL_ROOT_PASSWORD=anotherStrongPassword
+MYSQL_PASSWORD_FILE=.local-secrets/db-password
+MYSQL_ROOT_PASSWORD_FILE=.local-secrets/db-root-password
```
Just to clarify, didn't this fail with the `_FILE` suffix?

MasterOfCubesAU commented on src/lib/cogs/SixMans.py:
```diff
@@ -400,9 +400,13 @@ async def queue(self, interaction: discord.Interaction):
             self.queue.remove(interaction.user)
             await interaction.response.send_message(embed=self.bot.create_embed("SCUFFBOT SIX MANS", f"You have left the six mans queue.", None), ephemeral=True)
         else:
+            self.logger.info("a")
```
Remove these debug logger infos
